vector/src/main/java/im/vector/riotx/features/roommemberprofile/RoomMemberProfileFragment.kt

```diff
@@ -90,14 +95,25 @@ class RoomMemberProfileFragment @Inject constructor(
         matrixProfileAppBarLayout.addOnOffsetChangedListener(appBarStateChangeListener)
         viewModel.observeViewEvents {
             when (it) {
-                is RoomMemberProfileViewEvents.Loading               -> showLoading(it.message)
-                is RoomMemberProfileViewEvents.Failure               -> showFailure(it.throwable)
-                is RoomMemberProfileViewEvents.OnIgnoreActionSuccess -> Unit
-                is RoomMemberProfileViewEvents.StartVerification     -> handleStartVerification(it)
+                is RoomMemberProfileViewEvents.Loading                -> showLoading(it.message)
+                is RoomMemberProfileViewEvents.Failure                -> showFailure(it.throwable)
+                is RoomMemberProfileViewEvents.OnIgnoreActionSuccess  -> Unit
+                is RoomMemberProfileViewEvents.StartVerification      -> handleStartVerification(it)
+                is RoomMemberProfileViewEvents.ShareRoomMemberProfile -> handleShareRoomMemberProfile()
             }.exhaustive
         }
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.roomMemberProfileShareAction -> {
+                viewModel.handle(RoomMemberProfileAction.ShareRoomMemberProfile)
```

Comment:
It's maybe a bit over engineered to ask the view model to do that, and then the view model send a view event, which will be handled by this class.
We can keep this, but at least let the view model compute the link to share, and send it in the viewEvent